### PR TITLE
fix(rename and comments): package rename code and comment propagation

### PR DESCRIPTION
**Pull request by spaulins-usgs**
fixed package rename code (#1133). fixed problem with comments not propagating properly when data switched between internal and external (#1183).

### CODE REVIEW

File: flopy/mf6/mfmodel.py [hunk 1]
```diff
@@ -1151,6 +1152,76 @@ def remove_package(self, package_name):
...
+            New package name
+        """
+        try:
+            # get namefile package data
+            package_data = self.name_file.packages.get_data()
+        except MFDataException as mfde:
+            message = (
+                f"Error occurred while updating package names "
+                "from name file in model "
+                '"{self.name}".'
```
Comment (mwtoews) [inline flopy/mf6/mfmodel.py:1173]: this should be:
```
                "Error occurred while updating package names "
                f'from name file in model "{self.name}".'
```

File: flopy/mf6/mfpackage.py [hunk 2]
```diff
@@ -1637,9 +1641,12 @@ def filename(self, fname):
                 child_pkg_group._update_filename(self._filename, fname)
             except Exception:
                 print(
-                    "WARNING: Unable to update file name for parent"
-                    "package of {}.".format(self.name)
+                    f"WARNING: Unable to update file name for parent"
+                    "package of {self.package_name}."
```
Comment (mwtoews) [inline flopy/mf6/mfpackage.py:1645]: similar fix:
```
                    "WARNING: Unable to update file name for parent"
                    f"package of {self.package_name}."
```

File: flopy/mf6/mfpackage.py [hunk 3]
```diff
@@ -2385,6 +2392,61 @@ def create_package_dimensions(self):
...
+                try:
+                    exchange_recarray_data = (
+                        self.model_or_sim.name_file.exchanges.get_data()
+                    )
+                except MFDataException as mfde:
+                    message = (
+                        f"An error occurred while retrieving exchange "
+                        "data from the simulation name file.  The error "
+                        "occurred while processing gnc file "
+                        '"{self.filename}".'
```
Comment (mwtoews) [inline flopy/mf6/mfpackage.py:2426]:
```
                        "An error occurred while retrieving exchange "
                        "data from the simulation name file.  The error "
                        f'occurred while processing gnc file "{self.filename}".'
```

File: flopy/mf6/mfmodel.py [hunk 4]
```diff
@@ -1089,9 +1090,9 @@ def remove_package(self, package_name):
                 package_data = self.name_file.packages.get_data()
             except MFDataException as mfde:
                 message = (
-                    "Error occurred while reading package names "
+                    f"Error occurred while reading package names "
                     "from name file in model "
-                    '"{}".'.format(self.name)
+                    '"{self.name}"'
```
Comment (mwtoews) [inline flopy/mf6/mfmodel.py:1095]:
```
                    "Error occurred while reading package names "
                    f'from name file in model "{self.name}"'
```

File: flopy/mf6/mfmodel.py [hunk 5]
```diff
@@ -1128,9 +1129,9 @@ def remove_package(self, package_name):
                 self.name_file.packages.set_data(new_rec_array)
             except MFDataException as mfde:
                 message = (
-                    "Error occurred while setting package names "
-                    'from name file in model "{}".  Package name '
-                    "data:\n{}".format(self.name, new_rec_array)
+                    f"Error occurred while setting package names "
+                    'from name file in model "{self.name}".  Package name '
+                    "data:\n{new_rec_array}"
```
Comment (mwtoews) [inline flopy/mf6/mfmodel.py:1134]:
```
                    "Error occurred while setting package names "
                    f'from name file in model "{self.name}".  '
                    f"Package name data:\n{new_rec_array}"
```

File: flopy/mf6/mfmodel.py [hunk 6]
```diff
@@ -1151,6 +1152,76 @@ def remove_package(self, package_name):
...
+                None,
+                self._simulation_data.debug,
+            )
+        try:
+            self.name_file.packages.set_data(new_rec_array)
+        except MFDataException as mfde:
+            message = (
+                f"Error occurred while updating package names "
+                'from name file in model "{self.name}".  Package name '
+                "data:\n{new_rec_array}"
```
Comment (mwtoews) [inline flopy/mf6/mfmodel.py:1216]:
```
                "Error occurred while updating package names "
                f'from name file in model "{self.name}".  '
                f"Package name data:\n{new_rec_array}"
```

**Review by mwtoews — COMMENTED** [review]
A few f-strings need to be adjusted to work as intended.

File: flopy/mf6/mfmodel.py (hunk 1 above)
Comment (spaulins-usgs) [reply flopy/mf6/mfmodel.py]: thanks!  looks like Codacy caught that too.

**Pull request merged**